### ISSUE TITLE
Variables are not replaced in TaskGen rule unless preceded by whitespace 

### ISSUE THREAD

**Issue by franz-mokka** [waf-project/waf#1814]
I'm trying to create a task to compile a [ConTeXt][1] document. The compiler only accepts options with the shape `--option=value`, so I have to declare the task like this:
````python
TaskGen.declare_chain(
  name = 'context',
  rule = '${CONTEXT} --path=${SRC[0].parent} --result=${TGT} ${SRC}',
  shell = False,
  reentrant = False,
  ext_in = '.tex',
  ext_out = '.pdf',
)
````

but this doesn't replace `${SRC[0].parent}` and `${TGT}` in the generated command:

````python
['/usr/bin/context', '--path=${SRC[0].parent}', '--result=${TGT}', '../source/doc.tex']
````

If I use spaces instead of `=` in the rule, then the variables are replaced as expected, but ConTeXt is not able the recognize the options and their values.

[1]: http://wiki.contextgarden.net/Main_Page

**Comment by ita1024** [comment]
Pass `shell=False`.

Note: `SRC[0].parent` represents a Node object, not a string (use `SRC[0].parent.name` or `SRC[0].parent.abspath()`)

**Comment by franz-mokka** [comment]
I'm already using `shell=False`.

**Comment by ita1024** [comment]
I meant that you could use `shell=True` as a workaround until a fix is applied.

**Comment by franz-mokka** [comment]
Yeah, I figured you may have meant that after the fact, but I see you already fixed it on 9ed7d41, and I can confirm it's working correctly, thanks.

**Comment by ita1024** [comment]
The fix will be in waf 1.9.5 (in two or three weeks)
